src/docopt/dispatch.nim

```diff
@@ -0,0 +1,69 @@
+import docopt, macros, strutils, sequtils, typetraits
+
+macro runUserImplemented(x: typed, fallback: untyped): untyped =
+  let typ = x.getType
+  if typ.kind == nnkBracketExpr and typ[0].kind == nnkSym and $typ[0] == "typeDesc":
+    let call = newIdentNode("to" & $typ[1])
+    result = nnkCall.newTree(call, newIdentNode("v"))
+  else:
+    result = fallback
+
+proc to[T](v: Value): T =
+  when T is SomeInteger:
+    T(parseInt($v))
+  elif T is bool:
+    v.toBool
+  elif T is string:
+    $v
+  elif T is seq:
+    @v
+  else:
+    runUserImplemented(T):
+      {.error: "Invalid type \"" & $T & "\" in signature of dispatched procedure".}
+
+proc discardable(b: bool): bool {.discardable.} = b
+
+macro dispatchProc*(args: Table[string, Value], procedure: proc, conditions: varargs[string]): untyped =
```

Comment:
Outline the idea of what kind of code will be generated by looking at the arguments.